sdks/python/src/opik/evaluation/metrics/llm_judges/structured_output_compliance/metric.py

```diff
@@ -0,0 +1,96 @@
+import json
+from typing import Any, List, Optional, Union
+
+from opik.evaluation.metrics import base_metric, score_result
+from opik.evaluation.models import base_model, models_factory
+from pydantic import BaseModel
+
+from . import parser, template
+
+
+class StructuredOutputComplianceResponseFormat(BaseModel):
+    score: float
+    reason: List[str]
+
+
+class StructuredOutputCompliance(base_metric.BaseMetric):
+    """
+    Check if the output is a valid JSON and/or JSON-LD compatible.
+    Ideal solution would have Pydantic schema support.
+    """
+
+    def __init__(
+        self,
+        model: Optional[Union[str, base_model.OpikBaseModel]] = None,
+        name: str = "structured_output_compliance_metric",
+        track: bool = True,
+        project_name: Optional[str] = None,
+    ):
+        super().__init__(name=name, track=track, project_name=project_name)
+        self._init_model(model)
+
+    def _init_model(
+        self, model: Optional[Union[str, base_model.OpikBaseModel]]
+    ) -> None:
+        if isinstance(model, base_model.OpikBaseModel):
+            self._model = model
+        else:
+            self._model = models_factory.get(model_name=model)
+
+    def score(  # type: ignore
+        self,
+        output: str,
+        pydantic_schema: Optional[Any] = None,
+    ) -> score_result.ScoreResult:
+        """
+        Scores the structured output compliance of the output.
+        :param output: The output to score.
+        :param pydantic_schema: The pydantic schema to validate against.
+        """
+        json_schema_str = "No schema provided. Validate for JSON format only."
```

Comment:
Opik already has `IsJson` metric in heuristics namespace.
Let's not process the case when it's not given at all and make pydantic schema always required.